### PR TITLE
Rewrote the logic of interaction with the chat

### PR DESCRIPTION
**Pull request by nyood**
- Most of the phrases have been moved to the library
- The built-in ability to rebuild messages has been removed. This is now a job for the chat processors.
- Added built-in dynamic palette to not include third party libraries (colors.inc, morecolors.inc, etc.). See influx_chatcolor.txt for info about colors.
- Simple ranks support: SCP, CP, CCP by default

### CODE REVIEW

**Comment by TotallyMehis** [comment]
Cool! I've been wanting to redo the whole chat thing as it's really ugly.

I'll review the changes soon enough.

File: addons/sourcemod/scripting/influx_speclist.sp [hunk 1]
```diff
@@ -108,30 +108,30 @@ public Action Cmd_SpecList( int client, int args )
         {
             ++num;
             
-            Format( szMsg, sizeof( szMsg ), "%s%s{MAINCLR1}%N", szMsg, ( szMsg[0] ) ? "{CHATCLR}, " : "", i );
+            Format( szMsg, sizeof( szMsg ), "%s\x01\x04%N", szMsg, i );
```
Comment (TotallyMehis) [inline addons/sourcemod/scripting/influx_speclist.sp]: All hardcoded colors should just use the translation files. There needs to be a way to change the colors somewhere.

File: addons/sourcemod/translations/influx.phrases.txt [hunk 2]
```diff
@@ -17,23 +17,97 @@
...
+		"en"			"{G}ON{D}"
+	}
 	
 	//
 	// General
 	//
 	"INF_MUSTBERUNNING"
 	{
-		"en"			"{G}[INFLUX] {B}You must be running!"
+		"en"			"{D}You must be running!"
```
Comment (TotallyMehis) [inline addons/sourcemod/translations/influx.phrases.txt]: Why were the prefixes removed? `[TIMER]` with previous colors should be fine.

**Review by TotallyMehis — COMMENTED** [review]
Overall very good. Some changes I've wanted to do, too.

Some minor style differences can be fixed after, but the biggest one being how do server owners change their prefix and chat colors?
If the prefix is baked in the translations, that's fine, it's easy to replace all `[TIMER]` with `[MYPREFIX]`. However, the "chat color" and "highlight color" (CHATCLR and MAINCLR1 previously), I feel, are important features and can't be discarded for hardcoded ones. Can `influx_palette.ini` file have something like `{CHATCLR}` and `{HIGHLIGHT}` that points to these colors?

**Comment by nyood** [comment]
Most of the phrases are available in the general library for editing. 
I think that server owners are able to independently choose the highlight color and chat color using all possible color abbreviations from `configs / influx_palette.ini`.
 
Chat plugin prefix and phrases that were not included in the library, I will add soon

File: addons/sourcemod/scripting/influx_speclist.sp (hunk 1 above)
Comment (nyood) [reply addons/sourcemod/scripting/influx_speclist.sp]: Ok, I will add an item separator to `influx_palette.ini`

**Comment by nyood** [comment]
@TotallyMehis

**Comment by TotallyMehis** [comment]
I'm going to need a command / a link to a plugin for the CSS spectator chat alltalk before removing the feature.

**Comment by TotallyMehis** [comment]
Nvm, I was able to find one: https://forums.alliedmods.net/showthread.php?p=1593727

You can remove influx_specalltalk_css entirely.

File: addons/sourcemod/scripting/influx_core/natives_chat.sp [hunk 3]
```diff
@@ -1,134 +1,23 @@
...
-    if ( prefix )
-    {
-        Format( buffer, sizeof( buffer ), "%s %s%s", g_szChatPrefix, g_szChatClr, buffer );
-    }
-    
-    
-    // Add CSGO color fix.
-    if ( g_bIsCSGO )
-    {
-        Format( buffer, sizeof( buffer ), " \x01\x0B%s", buffer );
```
Comment (TotallyMehis) [inline addons/sourcemod/scripting/influx_core/natives_chat.sp:36]: Colors don't work properly without this.

File: addons/sourcemod/scripting/influx_simpleranks_chat.sp [hunk 4]
```diff
@@ -36,107 +70,91 @@ public APLRes AskPluginLoad2( Handle hPlugin, bool late, char[] szError, int err
...
+int MType;
+
+public void cc_proc_MsgBroadType(const int iType)
 {
-    if ( StrEqual( lib, "basecomm" ) ) g_bLib_BaseComm = false;
-    if ( StrEqual( lib, INFLUX_LIB_SILENT_CHATCMDS ) ) g_bLib_SilentChatCmds = false;
+    MType = iType;
 }
 
-public Action OnClientSayCommand( int client, const char[] szCommand, const char[] szMsg )
```
Comment (TotallyMehis) [inline addons/sourcemod/scripting/influx_simpleranks_chat.sp:68]: influx_simpleranks_chat won't work without a chat processor.

Add OnClientSayCommand and all it's funtionality back and return early if a chat processor library exists.

**Pull request closed without merging**